sim/hunter/kill_shot.go

```diff
@@ -0,0 +1,79 @@
+package hunter
+
+import (
+	"time"
+
+	"github.com/wowsims/cata/sim/core"
+	"github.com/wowsims/cata/sim/core/proto"
+)
+
+func (hunter *Hunter) registerKillShotSpell() {
+	if hunter.HasPrimeGlyph(proto.HunterPrimeGlyph_GlyphOfKillShot) {
+		icd := core.Cooldown{
+			Timer:    hunter.NewTimer(),
+			Duration: time.Second * 6,
+		}
+		ksReset := hunter.RegisterAura(core.Aura{
+			Label: "Kill Shot Glyph",
+			Icd: &icd,
+			Duration: time.Second,
+			OnGain: func(aura *core.Aura, sim *core.Simulation) {
+				hunter.KillShot.CD.Reset()
+			},
+		})
+		hunter.RegisterAura(core.Aura{
+			Label: "Kill Shot Glyph Activator",
+			Duration: core.NeverExpires,
+			OnReset: func(aura *core.Aura, sim *core.Simulation) {
+				aura.Activate(sim)
+			},
+			OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				if spell == hunter.KillShot {
+					if ksReset.Icd.IsReady(sim) {
+						ksReset.Icd.Use(sim)
+						ksReset.Activate(sim)
```

Comment:
you can just call `hunter.KillShot.CD.Reset()` instead of applying this aura. The only reason for an aura i would see is to show this in the metrics but maybe we can do this smarter as well as it feels a bit overkill to do it with an aura

Reply:
Had no idea Cooldowns worked without auras tbh. Then I can prolly change my counter auras in talents.go as well, take a look at the new implementation and see if I understood you correctly.